easydiffraction/calculators/cryspy.py

```diff
@@ -218,8 +220,25 @@ def assignCell_toCrystal(self, cell_name: str, crystal_name: str):
         crystal.cell = cell
 
     def createSpaceGroup(
-        self, key: str = "spacegroup", name_hm_alt: str = "P 1"
+        self, key: str = "spacegroup", name_hm_alt: str = "", it_code: Optional[str] = ""
     ) -> str:
+        if not name_hm_alt and not self.hm_symbol:
+            self.hm_symbol = "P 1"
+        if not name_hm_alt:
+            name_hm_alt = self.hm_symbol
+        else:
+            self.hm_symbol = name_hm_alt
+
+        if not it_code and not self.it_code:
+            self.it_code = "1"
```

Comment:
We should probably set the default value of `it_code` based on the current value of `hm_symbol`, since some space groups do not have a "1" in the list of possible it_codes. An example is "P n m a".

Reply:
ah yes, of course you are right. Knowing `hm_symbol` we can query for possible values of `it_code` and assign the first one.